plugins/modules/packaging/os/yum_versionlock.py

```diff
@@ -121,22 +121,25 @@ def main():
     versionlock_packages = yum_v.get_versionlock_packages()
 
     # Ensure versionlock state of packages
+    packages_list = []
     if state in ('present'):
         command = 'add'
         for single_pkg in packages:
             if single_pkg not in versionlock_packages:
                 if module.check_mode:
                     changed = True
                     continue
-                changed = yum_v.ensure_state(single_pkg, command)
+                packages_list.append(single_pkg)
+        changed = yum_v.ensure_state(packages_list, command)
```

Comment:
If `packages_list` is empty, this will cause the module to fail:
```suggestion
        if packages_list:
            changed = yum_v.ensure_state(packages_list, command)
```
(Same below.)

Reply:
https://github.com/ansible-collections/community.general/pull/1912/commits/aa80a3c8da141961979a2e1900c7f9aa0b1aeba4